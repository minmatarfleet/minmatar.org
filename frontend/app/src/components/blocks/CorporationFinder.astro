---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import Dialog from '@components/blocks/Dialog.astro';
import Input from '@components/blocks/Input.astro';
import Flexblock from '@components/compositions/Flexblock.astro';
import FixedFluid from '@components/compositions/FixedFluid.astro';
import VerticalCenter from './VerticalCenter.astro';

import corporations from '@json/hosted/corporations.json';

import { get_corporation_logo } from '@helpers/eve_image_server';

interface Props {
    [propName: string]: any;
}
---

<div
    class="[ corporations_finder ]"    
    x-data={`{
        filter: '',
        show_item(el) {
            const name_container = el.querySelector('span')
            const show_item = (
                this.filter != '' && name_container.textContent.toLowerCase().includes(this.filter.toLowerCase())
            )

            return show_item
        }
    }`}
>
    <Dialog
        x-bind:class="(corporations_finder_open ? 'open' : '')"
        x-bind:aria-hidden="corporations_finder_open == false"
        breakout={true}
        class="[ w-full max-w-[500px] ]"
        x-trap="corporations_finder_open"
        x-bind:inert="!corporations_finder_open"
    >
        <Input placeholder={t('find_corporation_placeholder')} x-model="filter" />

        <div class="[ relative ]">
            <Flexblock class="[ corporation-list ][ light-transparency w-full ]" gap='0'>
                {corporations.map((corporation) =>
                    <button
                        type="button"
                        corporation-finder-action="accept"
                        corporation-finder-data={JSON.stringify(corporation)}
                        x-show="show_item($el)"
                        x-on:click="(filter = '')"
                    >
                        <FixedFluid width='64px'>
                            <picture>
                                <img loading="lazy" src={get_corporation_logo(corporation.corporation_id, 64)} height="64" width="64" alt={`${corporation.name} ${t('profile_picture')}`} />
                            </picture>
                            <VerticalCenter>
                                <Flexblock gap='0' class="[ items-start ]">
                                    <span class="[ truncate ]">{corporation.name}</span>
                                    <small>{`${corporation.members_count} ${t('members_count').toLowerCase()}`}</small>
                                </Flexblock>
                            </VerticalCenter>
                        </FixedFluid>
                    </button>
                )}
            </Flexblock>
        </div>
    </Dialog>

    <div
        x-bind:class="(corporations_finder_open ? 'open' : '')"
        corporation-finder-action="close"
        x-on:click="(filter = '')"
    />
</div>

<style lang="scss">
    :global([corporation-finder-action] *) {
        pointer-events: none !important;
    }

    .dialog {
        top: 22%;
    }
    
    .corporation-list {
        position: absolute;
        overflow-y: auto;
        max-height: 320px;
        background-color: #171616;
        border: solid 1px var(--border-color);
        border-top: none;
    }

    [corporation-finder-action="close"] {
        position: fixed;
        inset: 0;
        z-index: var(--backdrop-z-index);
        display: none;

        &.open {
            display: block;
        }
    }

    [corporation-finder-action="accept"] {
        background-color: transparent;
        cursor: pointer;
        border: none;
        padding: 0;

        &:focus-within {
            outline: none;
            background-color: var(--component-background);
        }

        span {
            color: var(--highlight);
        }

        small {
            color: var(--foreground);
        }
        
        &:hover {
            background-color: var(--component-background);
        }
    }
</style>