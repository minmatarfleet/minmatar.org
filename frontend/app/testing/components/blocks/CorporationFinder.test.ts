import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test, vi } from "vitest";
import CorporationFinder from "@components/blocks/CorporationFinder.astro";

test("CorporationFinder defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(CorporationFinder, {});

  expect(result).toMatchSnapshot();
});
