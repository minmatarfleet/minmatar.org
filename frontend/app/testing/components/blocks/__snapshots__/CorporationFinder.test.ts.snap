// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CorporationFinder defaults 1`] = `
"<div class="[ corporations_finder ]" x-data="{
        filter: '',
        show_item(el) {
            const name_container = el.querySelector('span')
            const show_item = (
                this.filter != '' &#38;&#38; name_container.textContent.toLowerCase().includes(this.filter.toLowerCase())
            )

            return show_item
        }
    }" data-astro-cid-gajffecf> <div role="dialog" class="dialog solid-transparency breakout fixed [ w-full max-w-[500px] ]" x-bind:class="(corporations_finder_open ? 'open' : '')" x-bind:aria-hidden="corporations_finder_open == false" x-trap="corporations_finder_open" x-bind:inert="!corporations_finder_open" data-astro-cid-gajffecf="true"> <div class="wrapper" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-m);--wrapper_padding_inline_end: var(--space-m);--wrapper_padding_block: var(--space-m);--wrapper_max_width: 80rem;">  <div class="flex-block gap---space-m" data-mod style="--flex_block_gap: var(--space-m);">   <div class="[ cover-body ]">  <div class="[ input-group ]" data-astro-cid-ovtlrinh style="--width: 100%;">  <input placeholder="Type corporation name to find" x-model="filter" data-astro-cid-gajffecf="true" class="astro-ovtlrinh" data-astro-cid-ovtlrinh style="--width: 100%;"> </div>   <div class="[ relative ]" data-astro-cid-gajffecf> <div class="flex-block [ corporation-list ][ light-transparency w-full ] gap-0" data-astro-cid-gajffecf="true" data-mod style="--flex_block_gap: 0;"> <button type="button" corporation-finder-action="accept" corporation-finder-data="{&#34;id&#34;:1,&#34;name&#34;:&#34;Rattini Tribe&#34;,&#34;members_count&#34;:316,&#34;corporation_id&#34;:98726134}" x-show="show_item($el)" x-on:click="(filter = '')" data-astro-cid-gajffecf> <div class="fixed-fluid" data-astro-cid-gajffecf="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture data-astro-cid-gajffecf> <img loading="lazy" src="https://images.evetech.net/corporations/98726134/logo?size=64" height="64" width="64" alt="Rattini Tribe profile picture" data-astro-cid-gajffecf> </picture> <div class="!flex !items-center" data-astro-cid-gajffecf="true">  <div class="flex-block [ items-start ] gap-0" data-astro-cid-gajffecf="true" data-mod style="--flex_block_gap: 0;">  <span class="[ truncate ]" data-astro-cid-gajffecf>Rattini Tribe</span> <small data-astro-cid-gajffecf>316 members count</small>  </div>   </div>  </div>   </button><button type="button" corporation-finder-action="accept" corporation-finder-data="{&#34;id&#34;:2,&#34;name&#34;:&#34;Black Omega Security&#34;,&#34;members_count&#34;:356,&#34;corporation_id&#34;:263585335}" x-show="show_item($el)" x-on:click="(filter = '')" data-astro-cid-gajffecf> <div class="fixed-fluid" data-astro-cid-gajffecf="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture data-astro-cid-gajffecf> <img loading="lazy" src="https://images.evetech.net/corporations/263585335/logo?size=64" height="64" width="64" alt="Black Omega Security profile picture" data-astro-cid-gajffecf> </picture> <div class="!flex !items-center" data-astro-cid-gajffecf="true">  <div class="flex-block [ items-start ] gap-0" data-astro-cid-gajffecf="true" data-mod style="--flex_block_gap: 0;">  <span class="[ truncate ]" data-astro-cid-gajffecf>Black Omega Security</span> <small data-astro-cid-gajffecf>356 members count</small>  </div>   </div>  </div>   </button><button type="button" corporation-finder-action="accept" corporation-finder-data="{&#34;id&#34;:2,&#34;name&#34;:&#34;Balls Deep Inc.&#34;,&#34;members_count&#34;:43,&#34;corporation_id&#34;:98252708}" x-show="show_item($el)" x-on:click="(filter = '')" data-astro-cid-gajffecf> <div class="fixed-fluid" data-astro-cid-gajffecf="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture data-astro-cid-gajffecf> <img loading="lazy" src="https://images.evetech.net/corporations/98252708/logo?size=64" height="64" width="64" alt="Balls Deep Inc. profile picture" data-astro-cid-gajffecf> </picture> <div class="!flex !items-center" data-astro-cid-gajffecf="true">  <div class="flex-block [ items-start ] gap-0" data-astro-cid-gajffecf="true" data-mod style="--flex_block_gap: 0;">  <span class="[ truncate ]" data-astro-cid-gajffecf>Balls Deep Inc.</span> <small data-astro-cid-gajffecf>43 members count</small>  </div>   </div>  </div>   </button> </div>  </div>  </div>   </div>   </div>   </div>  <div x-bind:class="(corporations_finder_open ? 'open' : '')" corporation-finder-action="close" x-on:click="(filter = '')" data-astro-cid-gajffecf></div> </div> "
`;
